doc/src/manual/environment-variables.md

```diff
@@ -16,12 +16,17 @@ including those which include `JULIA` in their names.
 
 !!! note
 
-    Some variables, such as [`JULIA_NUM_THREADS`](@ref JULIA_NUM_THREADS) and
-    [`JULIA_PROJECT`](@ref JULIA_PROJECT), need to be set before Julia
-    starts. Adding these to `~/.julia/config/startup.jl` is too late in the startup process.
+    It is recommended to avoid changing environment variables during runtime,
+    such as within a `~/.julia/config/startup.jl`.
 
-    Also `__init__()` functions of modules in the sysimage will not see environment variables
-    that are set in `startup.jl` as they are run beforehand.
+    One reason is that some julia language variables, such as [`JULIA_NUM_THREADS`](@ref JULIA_NUM_THREADS)
+    and [`JULIA_PROJECT`](@ref JULIA_PROJECT), need to be set before Julia starts.
+
+    Similarly, `__init__()` functions of modules in the sysimage are run before `startup.jl`,
+    so setting environment variables in a `startup.jl` may be too late for user code.
```

Comment:
I am still not a fan of the "in the sysimage", I think the more common scenario is the `init` functions of modules being used from startup.jl.

Reply:
A user can see the order of things in a startup.jl but they may not be aware that sysimage module units are run before startup.jl is run. 

I wanted to add this specifically as it caught me out.

Reply:
How about 
```suggestion
    Similarly, `__init__()` functions of user modules in the sysimage (via PackageCompiler) are 
    run before `startup.jl`, so setting environment variables in a `startup.jl` may be too late for 
    user code.
```